app/models/step.rb

```diff
@@ -27,13 +27,19 @@ def request_broken_links
 private
 
   def collect_broken_links
+    return [] if batch_link_report.empty?
+
     batch_link_report.links.keep_if do |link|
       link.fetch('status') == 'broken'
     end
   end
 
   def batch_link_report
-    @batch_link_report ||= Services.link_checker_api.get_batch(batch_link_report_id)
+    begin
+      @batch_link_report ||= Services.link_checker_api.get_batch(batch_link_report_id)
+    rescue GdsApi::HTTPServerError
+      []
```

Comment:
It feels a little weird to use an empty array here as it sort of implies that a fully populated `@batch_link_report` would also be an array, which I don't think it is.  Maybe use `nil` instead?